generate/templates/manual/src/thread_pool.cc

```diff
@@ -672,12 +674,10 @@ namespace nodegit {
       jsThreadCallbackQueue.pop();
     }
 
-    // NOTE We are deliberately leaking this pointer because `async` cleanup in
-    // node has not completely landed yet. Trying to cleanup this pointer
-    // is probably not worth the fight as it's very little memory lost per context
-    // When all LTS versions of node and Electron support async cleanup, we should
-    // be heading back to cleanup this
-    uv_close((uv_handle_t *)jsThreadCallbackAsync, nullptr);
+    uv_close(reinterpret_cast<uv_handle_t *>(jsThreadCallbackAsync), [](uv_handle_t *handle) {
+      auto threadPoolImpl = static_cast<ThreadPoolImpl *>(handle->data);
+      delete threadPoolImpl->currentContext;
```

Comment:
I'm not really a fan of the threadpool deleting the context like this. Could we add a completion callback to the shutdown routine instead?